apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloParseInterceptor.java

```diff
@@ -81,12 +82,15 @@ public void interceptAsync(@NotNull final InterceptorRequest request, @NotNull A
     String cacheKey = httpResponse.request().header(HttpCache.CACHE_KEY_HEADER);
     if (httpResponse.isSuccessful()) {
       try {
-        OperationResponseParser parser = new OperationResponseParser(operation, responseFieldMapper, scalarTypeAdapters,
-            normalizer);
-        Response parsedResponse = parser.parse(httpResponse.body().source())
+        final OperationResponseParser parser = new OperationResponseParser(operation, responseFieldMapper, scalarTypeAdapters, normalizer);
+        final OkHttpExecutionContext httpExecutionContext = new OkHttpExecutionContext(httpResponse);
+        Response parsedResponse = parser.parse(httpResponse.body().source());
+        parsedResponse = parsedResponse
             .toBuilder()
             .fromCache(httpResponse.cacheResponse() != null)
+            .executionContext(parsedResponse.getExecutionContext().set(OkHttpExecutionContext.KEY, httpExecutionContext))
```

Comment:
What types of objects do we plan to put in the context?

CoroutinesContext has `Dispatcher`, `Job`, `Name`, maybe more
ExecutionContext will have `OkHttpExecutionContext` and what else ? Do we plan to put dispatchers/executors down there too ?

Reply:
For now we put just http context, but in future for new runtime we are going to put cache policy, cache headers, cache normalization records etc. Also any implementation of future request executor (now it's called Apollo interceptor) can put whatever it wants.